components/backup/src/service.rs

```diff
@@ -137,13 +147,35 @@ where
 
         ctx.spawn(send_task);
     }
+
+    /// The new method for preparing a disk snapshot backup.
+    /// Generally there will be some steps for the client to do:
+    /// 1. Establish a `prepare_snapshot_backup` connection.
+    /// 2. Send a initial `UpdateLease`. And we should update the lease
+    /// periodically.
+    /// 3. Send `WaitApply` to each leader peer in this store.
+    /// 4. Once `WaitApply` for all regions have done, we can take disk
+    /// snapshot.
+    /// 5. Once all snapshots have been taken, send `Finalize` to stop.
+    fn prepare_snapshot_backup(
+        &mut self,
+        ctx: grpcio::RpcContext<'_>,
+        stream: grpcio::RequestStream<PrepareSnapshotBackupRequest>,
+        sink: grpcio::DuplexSink<PrepareSnapshotBackupResponse>,
+    ) {
+        let l = StreamHandleLoop::new(self.snap_br_env.clone());
+        ctx.spawn(async move {
+            if let Err(err) = l.run(stream, sink.into()).await {
+                warn!("stream closed; perhaps a problem cannot be retried happens"; "reason" => ?err);
+            }
```

Comment:
It's better to offload complex event loop to a future runtime, in case grpc threads are blocked by the event loop accidently.

Reply:
Done, I will make it share with the IO pool of backing up.